crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -1538,18 +1540,6 @@ where
         }
     }
 
-    fn visit_match_case(&mut self, match_case: &'ast ast::MatchCase) {
-        debug_assert!(self.current_match_case.is_none());
-        self.current_match_case = Some(CurrentMatchCase::new(&match_case.pattern));
-        self.visit_pattern(&match_case.pattern);
-        self.current_match_case = None;
-
-        if let Some(expr) = &match_case.guard {
-            self.visit_expr(expr);
-        }
-        self.visit_body(&match_case.body);
-    }
```

Comment:
This isn't used anywhere else, so I updated the call site to only call it once and then inlined it.  That way I didn't need to add a field to pass the constraint ID back up to where we're iterating through all of the cases.